be/src/formats/orc/apache-orc/c++/src/sargs/SargsApplier.cc

```diff
@@ -86,10 +86,14 @@ bool SargsApplier::pickRowGroups(uint64_t rowsInStripe, const std::unordered_map
     mNextSkippedRows.resize(groupsInStripe);
     mTotalRowsInStripe = rowsInStripe;
 
+    // NOTE(yanz): We can skip following steps if rowIndexes.empty.
+    // Because we have to set right values into `mNextSkippedRows` and
+    // those values will be used in `hasSelectedFrom`.
+
     // row indexes do not exist, simply read all rows
-    if (rowIndexes.empty()) {
-        return true;
-    }
+    // if (rowIndexes.empty()) {
```

Comment:
delete ?

Reply:
this code is from apache/orc. better to leave it here for cross reference.